undertow/src/main/java/org/wildfly/extension/undertow/WebServerService.java

```diff
@@ -50,7 +50,7 @@ public int getPort(final String protocol, final boolean secure) {
         Map<String, UndertowListener> listeners = getListenerMap();
         UndertowListener listener = null;
         for (String p : listeners.keySet()) {
-            if (protocol.toLowerCase().contains(p)) {
+            if (protocol.toLowerCase().contains(p) || listeners.size() == 1) {
```

Comment:
The CommonWebServer javadoc needs to specify the contract here, as it's not obvious what the expected behavior is.

This '|| listeners.size() == 1' doesn't seem precise enough. What is there is only an AJP listener?  What if there's only HTTPS and 'secure' is false?

Reply:
`|| listeneres.size() == 1` will fix the `secure == false` when there's only a HTTPS listener issue. Right now it fails.

Is there's only an AJP listener than it would throw an exception, as expected since it's not a HTTP-family listener.